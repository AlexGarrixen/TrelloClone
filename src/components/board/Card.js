import React from 'react';
import { Draggable } from 'react-smooth-dnd';
import Card from '../ui/Card';
import Badge from '../ui/Badge';
import Icons from '../icons';

const CardItem = () => (
  <Draggable>
    <Card className='board-list-card' title='Title'>
      <ul className='board-list-card__grid-labels'>
        <Badge>Badge</Badge>
        <Badge>Badge</Badge>
        <Badge>Badge</Badge>
      </ul>
      <div className='board-list-card__footer'>
        <p className='board-list-card__num-comments'>
          <Icons.Comment className='mr-2' /> 2
        </p>
        <p className='board-list-card__num-attachments'>
          <Icons.PaperClip className='mr-2' /> 1
        </p>
      </div>
    </Card>
  </Draggable>
);

export default CardItem;
