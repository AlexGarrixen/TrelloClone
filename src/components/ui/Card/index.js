import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';

const Card = ({
  children,
  component: Component = 'article',
  className,
  picture,
  title,
  ...other
}) => (
  <Component className={clsx('card', className)} {...other}>
    <div
      className='card__picture'
      style={{ backgroundImage: `url(${picture})` }}
    />
    <h2 className='card__title'>{title}</h2>
    {children}
  </Component>
);

Card.propTypes = {
  title: PropTypes.string,
  picture: PropTypes.string,
  children: PropTypes.node,
  component: PropTypes.elementType,
};

export default Card;
