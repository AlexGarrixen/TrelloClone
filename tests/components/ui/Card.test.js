import React from 'react';
import { mount } from 'enzyme';
import Card from '../../../src/components/ui/Card';

describe('<Card />', () => {
  test('Render Success', () => {
    const wrapper = mount(<Card>Example</Card>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.card__picture').exists()).toBe(true);
    expect(wrapper.find('.card__title').exists()).toBe(true);
  });

  test('Test with default props', () => {
    const wrapper = mount(<Card>Example</Card>);
    const element = wrapper.find('.card');
    const elementPicture = wrapper.find('.card__picture');
    const elementTitle = wrapper.find('.card__picture');

    expect(element.text()).toBe('Example');
    expect(element.type()).toBe('article');
    expect(elementTitle.text()).toBe('');
    expect(elementPicture.props().style.backgroundImage).toBe(`url(undefined)`);
  });

  test('Test with prop component', () => {
    const wrapper = mount(<Card component='a'>Example</Card>);
    const element = wrapper.find('.card');

    expect(element.type()).toBe('a');
  });

  test('Test with prop picture', () => {
    const wrapper = mount(<Card picture='https://example.com'>Example</Card>);
    const elementPicture = wrapper.find('.card__picture');
    const backgroundImage = elementPicture.props().style.backgroundImage;

    expect(backgroundImage).toBe('url(https://example.com)');
  });

  test('Test with prop title', () => {
    const wrapper = mount(<Card title='Title example'>Example</Card>);
    const elementTitle = wrapper.find('.card__title');

    expect(elementTitle.text()).toBe('Title example');
  });
});
