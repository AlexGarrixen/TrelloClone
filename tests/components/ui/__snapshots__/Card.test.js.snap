// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Render Success 1`] = `
<article
  className="card"
>
  <div
    className="card__picture"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
  <h2
    className="card__title"
  />
  Example
</article>
`;
